sandbox/apparmor/apparmor.go

```diff
@@ -374,6 +381,39 @@ func tryAppArmorParserFeature(parser, rule string) bool {
 	return true
 }
 
+// UpdateHomedirsTunable sets the AppArmor HOMEDIRS tunable to the list of the
+// specified directories. This directly affects the value of the AppArmor
+// @{HOME} variable. See the "/etc/apparmor.d/tunables/home" file for more
+// information.
+func UpdateHomedirsTunable(homedirs []string) error {
+	homeTunableDir := filepath.Join(ConfDir, "tunables/home.d")
+	tunableFilePath := filepath.Join(homeTunableDir, "snapd")
+
+	// If the file is not there and `homedirs` is empty, do nothing; this is
+	// not just an optimisation, but a necessity in Ubuntu Core: the
+	// /etc/apparmor.d/ tree is read-only, and attempting to create the file
+	// would generate an error.
+	if len(homedirs) == 0 && !osutilFileExists(tunableFilePath) {
+		return nil
+	}
+
+	if err := osMkdirAll(homeTunableDir, 0755); err != nil {
+		return fmt.Errorf("cannot create AppArmor tunable directory: %v", err)
+	}
+
+	contents := &bytes.Buffer{}
+	fmt.Fprintln(contents, "# Generated by snapd -- DO NOT EDIT!")
+	if len(homedirs) > 0 {
+		contents.Write([]byte("@{HOMEDIRS}+="))
+		separator := ""
+		for _, dir := range homedirs {
+			fmt.Fprintf(contents, `%s"%s"`, separator, dir)
+			separator = " "
+		}
```

Comment:
Should we add something like: `contents.Write([]byte{"\n"})` here to add a final \n?